File: project/iosApp/ios/ExpenseStatisticsMobile/ClipboardModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ClipboardModule, NSObject)

RCT_EXTERN_METHOD(copyText:(NSString *)text
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: project/iosApp/ios/ExpenseStatisticsMobile/ClipboardModule.swift
import UIKit
import React

@objc(ClipboardModule)
class ClipboardModule: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    true
  }

  @objc(copyText:resolver:rejecter:)
  func copyText(
    _ text: NSString,
    resolver resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {
    UIPasteboard.general.string = text as String
    resolve(nil)
  }
}
